src/libs/ck-libs/ampi/ampi.C

```diff
@@ -11761,8 +11761,8 @@ int AMPI_GPU_Iinvoke(cudaStream_t stream, MPI_Request *request)
 
   // A callback that completes the corresponding request
   CkCallback *cb = new CkCallback(&AMPI_GPU_complete, newreq);
```

Comment:
Can this just be created on the stack instead of heap allocated?

Reply:
I think the problem is `hapiWorkRequestSetCallback` and `hapiAddCallback` used in AMPI are declared as `AMPI_CUSTOM_FUNC` in `hapi_functions.h`, and Evan mentioned that these functions should not take `CkCallback` as a parameter. That's why these are taking a `void *` instead and require `CkCallback` to be allocated on the heap.

Reply:
You don't have to allocate structures on the heap to pass them as `void *`. You can create `cb` as a normal local variable and pass `&cb` into the function call.

Reply:
@evan-charmworks Good point, I didn't realize I could do that because there is a delete call afterwards anyways.

Reply:
@evan-charmworks @stwhite91 Just pushed a commit to fix this issue